django_project/frontend/src/store/authSlice.ts

```diff
@@ -88,30 +98,35 @@ export const checkLoginStatus = () => async (dispatch: AppDispatch) => {
         user: response.data.user,
         token: token,
       }));
+      return;
     } catch (error) {
+      console.warn("Token validation failed, falling back to user info check.");
+    }
+  }
+
+  // Fallback: Check user info if the token validation fails
+  try {
+    setCSRFToken();
+    const response = await axios.post("/api/user-info/", {
+      credentials: "include",
+    });
+
+    if (response.data.is_authenticated) {
+      dispatch(authSlice.actions.setAuthenticationStatus(true));
```

Comment:
@tinashechiraya when the code use the fallback, don't we need to fetch the user data as well? so other places that are using user state can still work.

Reply:
Hi @danangmassandy no it won't be needed and at the moment we are not using user data to that extend

Reply:
@danangmassandy resolved